src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/swap_convert_transpose.cpp

```diff
@@ -32,11 +32,12 @@ ov::intel_cpu::SwapConvertTranspose::SwapConvertTranspose() {
         transposeInputs[0] = convert->input_value(0);
         auto newTranspose = transpose->clone_with_new_inputs(transposeInputs);
         ngraph::copy_runtime_info(transpose, newTranspose);
-        newTranspose->set_friendly_name(transpose->get_friendly_name());
+        newTranspose->set_friendly_name(convert->get_friendly_name());
```

Comment:
why should we use `convert` name if outputs of `convert` and new `transpose` are different?

Reply:
The transformation broke the output names on new API 2.0 for CPU Plugin. Let's take a look at the simple test `./ov_cpu_func_tests --gtest_filter=smoke_Transpose2D/TransposeLayerTest.CompareWithRefs/IS=\(2.10\)_inputOrder=\(\)_netPRC=I8_inPRC=UNSPECIFIED_outPRC=UNSPECIFIED_inL=ANY_outL=ANY_trgDev=CPU*`
We have the following transformation pipeline (`Convert` is inserted after new preprocessing):
![image](https://github.com/openvinotoolkit/openvino/assets/43129309/3b1100ba-f7ed-44ba-88b4-9f831823c87a)
The model expects output with the name `Transpose_3` but after the pass `SwapConvertTranspose` the model has output node with name `Convert_1696`. To avoid these cases we should preserve the output name after transformation.

Thank you for the question! 🙂